be/src/vec/exec/scan/scanner_scheduler.cpp

```diff
@@ -384,10 +384,15 @@ void ScannerScheduler::_scanner_scan(ScannerScheduler* scheduler, ScannerContext
             ctx->return_free_block(std::move(block));
         } else {
             if (!blocks.empty() && blocks.back()->rows() + block->rows() <= state->batch_size()) {
-                status = vectorized::MutableBlock(blocks.back().get()).merge(*block);
+                auto rows = blocks.back()->rows() + block->rows();
+                vectorized::MutableBlock mutable_block(blocks.back().get());
+                status = mutable_block.merge(*block);
                 if (!status.ok()) {
                     break;
                 }
+                if (!blocks.back().get()->is_valid(rows)) {
```

Comment:
Is there any situation that it's valid that block has columns with different size? If not, maybe we can do this check in merging block.

Reply:
the key point is that we reassign value to block from mutable block, and where to do check i think are both ok